lambdas/validate-oauth-callback/src/main/java/uk/gov/di/ipv/core/validateoauthcallback/ValidateOAuthCallbackHandler.java

```diff
@@ -105,28 +107,49 @@ public Map<String, Object> handleRequest(CriCallbackRequest callbackRequest, Con
 
         IpvSessionItem ipvSessionItem = null;
         CriOAuthSessionItem criOAuthSessionItem = null;
+        ClientOAuthSessionItem clientOAuthSessionItem;
 
         try {
             String ipvSessionId = callbackRequest.getIpvSessionId();
-            if (ipvSessionId == null) {
+            String criOAuthSessionId = callbackRequest.getState();
+
+            if (ipvSessionId != null) {
+                ipvSessionItem = ipvSessionService.getIpvSession(ipvSessionId);
+            } else if (criOAuthSessionId == null) {
                 throw new HttpResponseExceptionWithErrorBody(
-                        HttpStatus.SC_BAD_REQUEST, ErrorResponse.MISSING_IPV_SESSION_ID);
+                        HttpStatus.SC_BAD_REQUEST, ErrorResponse.MISSING_OAUTH_STATE);
+            } else {
+                ipvSessionItem =
+                        ipvSessionService
+                                .getIpvSessionByCriOAuthSessionId(criOAuthSessionId)
```

Comment:
I think, We should be careful for performance. We can add to the new key for this property.